tools/docker/Run-ContainerTests.ps1

```diff
@@ -0,0 +1,278 @@
+      }
+      else {
+         Write-Verbose "image $Tag already exists with id $dockerImageId"
+      }
+   }
+}
+
+function Start-DockerVSTeamTests {
+   [CmdletBinding()]
+   param (
+      # Name of the container to run
+      [Parameter(Mandatory = $true)]
+      [string]
+      $Container,
+      # volume mapping string
+      # see: https://docs.docker.com/storage/volumes/#start-a-container-with-a-volume
+      [Parameter(Mandatory = $true)]
+      [string]
+      $Volume,
+      # default directoy when the container is started
+      [Parameter(Mandatory = $true)]
+      [string]
+      $DefaultWorkDir,
+      # Image to start
+      [Parameter(Mandatory = $true)]
+      [string]
+      $Image,
+      # choose which shell to start
+      [Parameter(Mandatory = $false)]
+      [ValidateSet("pwsh", "powershell")]
+      [string]
+      $Shell = "pwsh",
+      # Use if powershell should wait for the container to exit
+      [Parameter(Mandatory = $false)]
+      [Switch]
+      $Wait,
+      # Open another powershell session to show the logs from the container
+      [Parameter(Mandatory = $false)]
+      [Switch]
+      $FollowLogs
+   )
+
+   begin {
+
+      # using a script block here to have syntax checking and highlightning.
+      # Later it is converted to a string to start the container with it
+      $pesterBuild = {
+
+         .\Build-Module.ps1 -installDep;
+         $null = Import-Module Pester;
+
+         $pesterArgs = [PesterConfiguration]::Default;
+         $pesterArgs.Run.Exit = $true;
+         $pesterArgs.Run.Path = '.\unit';
+         $pesterArgs.Run.PassThru = $true;
+         $pesterArgs.Output.Verbosity = 'Normal';
+         $pesterArgs.TestResult.Enabled = $true;
+         $pesterArgs.TestResult.OutputPath = '#Container#_result.xml'
+
+         Invoke-Pester -Configuration $pesterArgs;
+
+         # exist with PowerShells last exist code for docker.
+         # without deliverate exit code the -Wait switch could cause to wait indefinetely
+         Exit $LASTEXITCODE
+      }
+   }
+
+   process {
+
+      $containerId = docker ps --all --filter name=$Container -q
+      $containerIsRunning = $null -ne (docker ps --filter name=$Container -q)
+
+      if ($containerIsRunning) {
+         docker stop $containerId
+      }
+
+      if ($containerId) {
+         docker rm $Container
+      }
+
+      $psCommandString = ($pesterBuild.ToString()) -replace '#Container#', $Container
+
+      docker run `
+         -dit `
+         --name  $Container `
+         --volume $Volume `
+         -w $DefaultWorkDir `
+         $Image `
+         $Shell -Command $psCommandString
+
+      if ($FollowLogs) {
+         $output = (docker exec -it $Container $Shell -c '$PSVersionTable | ConvertTo-Json -Compress;') -join ''
+         $outputFirst = $output.IndexOf('{')
+         $ouputLast = $output.LastIndexOf('}')
+         $versiontable = $output.Substring($outputFirst, $ouputLast+1-$outputFirst) | ConvertFrom-Json -Depth 50
```

Comment:
ConvertFrom-Json does not have a -Depth. Only ConvertTo-Json has that.

Reply:
Fixed it. Didnt get an error before and VSCode seems to suggest that as a parameter.